sklearn/externals/_arff.py

```diff
@@ -171,6 +172,18 @@
 _RE_SPARSE_LINE = re.compile(r'^\s*\{.*\}\s*$', re.UNICODE)
 _RE_NONTRIVIAL_DATA = re.compile('["\'{}\\s]', re.UNICODE)
 
+ArffDataType = Tuple[List, ...]
+
+if typing.TYPE_CHECKING:
```

Comment:
According to [TYPE_CHECKING docs](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) this is used for performance reasons. If this is case, let's have a comment here.

Reply:
No the reason is that https://pypi.org/project/typing-extensions/  (imported below) is a mypy dependency so we can only import it if we are checking for types and therefore it is installed.